### PR TITLE
feat: update entity

### PR DESCRIPTION
**Pull request by fes300**
added funcionality to update an entity

### CODE REVIEW

File: src/infrastructure/catalog/entities/tag.ts [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
-interface _Tag<T extends string, B = never> {
-  id: string;
-  name: string;
-  type: T;
-  state: EntityState;
-  body: B;
-}
-
-export type Tag = _Tag<"in-evidence", Partial<{ order: number }>>;
+export const Tag = D.union(
```
Comment (giovannigiordano) [inline src/infrastructure/catalog/entities/tag.ts]: why union?

File: src/infrastructure/catalog/index.ts [hunk 2]
```diff
@@ -184,13 +349,25 @@ export class CatalogInfrastructure
...
+
+  updateEntity = <E extends Entity>(
+    entity: {
+      type: E["type"];
+      id: string;
+    },
+    updater: EntityUpdate<E["body"]>
+  ) => {
+    const rowsToUpdate = this.getEntityUpdateOperations(
+      this.getDynamoId("it", entity.type, entity.id),
```
Comment (giovannigiordano) [inline src/infrastructure/catalog/index.ts]: Should the literal "it" be a variable?

**Review by giovannigiordano — APPROVED** [review]

File: src/infrastructure/catalog/entities/tag.ts (hunk 1 above)
Comment (fes300) [reply src/infrastructure/catalog/entities/tag.ts]: I was anticipating the existence of other kinds of tags (ex. tags that become banners in app)

File: src/infrastructure/catalog/index.ts (hunk 2 above)
Comment (fes300) [reply src/infrastructure/catalog/index.ts]: yeah it should :)
also @stefanew pointed that out, I'll change it

**Pull request merged**